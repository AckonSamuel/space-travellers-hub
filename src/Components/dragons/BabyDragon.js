import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { addReservation, cancelReservation } from '../../redux/dragons/dragons.redux';
import './Dragons.css';

const BabyDragon = ({
  name, type, description, image, reserved, id,
}) => {
  const dispatch = useDispatch();
  const handleAddReservation = () => {
    dispatch(addReservation(id));
  };

  const handleCancelReservation = () => {
    dispatch(cancelReservation(id));
  };

  return (
    <li>
      <div className="image_container">
        <img src={image} alt={name} className="dragon_image" />
      </div>
      <div className="description-container">
        <h3 className="title">{name}</h3>
        <h5 className="type">
          (
          {type}
          )
        </h5>
        <p>
          {reserved ? <span className="span">Reserved</span> : ''}
          {description}
        </p>
        {reserved ? (
          <button type="button" className="reserve cancel" onClick={handleCancelReservation}>Cancel Reservation</button>
        ) : (
          <button type="button" className="reserve" onClick={handleAddReservation}>Reserve Dragon</button>
        )}
      </div>
    </li>
  );
};

BabyDragon.propTypes = {
  description: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  image: PropTypes.node.isRequired,
};

export default BabyDragon;
