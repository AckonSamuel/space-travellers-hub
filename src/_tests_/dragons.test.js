import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer'
import store from '../redux/configureStore';
import BabyDragon from '../Components/dragons/BabyDragon';

describe(' Test for dragon component', () => {
    it('Check if dragon component renders', () => {
        const dragon = {
         name: 'Dragon 1',
         type: 'capsule', 
         description: 'Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California. Dragon is launched into space by the SpaceX Falcon 9 two-stage-to-orbit launch vehicle. The Dragon spacecraft was originally designed for human travel, but so far has only been used to deliver cargo to the International Space Station (ISS)',
         flickr_image: [ '<img src="https://i.imgur.com/9fWdwNv.jpg" alt="Dragon 1" class="dragon_image">' ] , 
         reserved: false, 
         id: 'dragon 1',
        };
        const dragonTree = renderer.create(
            <Provider store={store}>
                <BabyDragon
                key={dragon.id}
                description={dragon.description}
                name={dragon.name}
                reserved={dragon.reserved}
                id={dragon.id}
                image={dragon.flickr_image[0]}
                type={dragon.type}
                />
            </Provider>
        );
        expect(dragonTree).toMatchSnapshot();
    })
})
