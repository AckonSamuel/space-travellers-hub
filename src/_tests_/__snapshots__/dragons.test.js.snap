// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Test for dragon component Check if dragon component renders 1`] = `
<li>
  <div
    className="image_container"
  >
    <img
      alt="Dragon 1"
      className="dragon_image"
      src="<img src=\\"https://i.imgur.com/9fWdwNv.jpg\\" alt=\\"Dragon 1\\" class=\\"dragon_image\\">"
    />
  </div>
  <div
    className="description-container"
  >
    <h3
      className="title"
    >
      Dragon 1
    </h3>
    <h5
      className="type"
    >
      (
      capsule
      )
    </h5>
    <p>
      Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California. Dragon is launched into space by the SpaceX Falcon 9 two-stage-to-orbit launch vehicle. The Dragon spacecraft was originally designed for human travel, but so far has only been used to deliver cargo to the International Space Station (ISS)
    </p>
    <button
      className="reserve"
      onClick={[Function]}
      type="button"
    >
      Reserve Dragon
    </button>
  </div>
</li>
`;
